tools/schemapi/schemapi.py

```diff
@@ -584,7 +736,10 @@ def indent_second_line_onwards(message: str, indent: int = 4) -> str:
         error_messages: list[str] = []
         # Only show a maximum of 3 errors as else the final message returned by this
         # method could get very long.
-        for errors in list(self._errors.values())[:3]:
+        # ^^^^^^^^^^
+        # CORRECTION: Only show 3 **json_paths**
+
+        for errors in islice(_group_errors_by_json_path(self._errors).values(), 3):
```

Comment:
The call to `_group_errors_by_json_path` is purely to satisfy the additional error message format handling.

Ideally recollecting into a `dict[str, list[ValidationError]` would be avoided, especially since:
- Only the first 3 groups are needed
- The keys (json path) are discarded